### PR TITLE
ref(slack): Refactor Unlink Identity View

### PR DESCRIPTION
**Pull request by iamrajjoshi**
https://www.notion.so/sentry/Slack-ff5b8ab4fd1e4760ab829438ce97ce15

Addresses `bot-2` task. 

Here, I refactored the view to have two separate methods for get and post. I added bunch of logging and metrics & improved tests.

### CODE REVIEW

File: src/sentry/integrations/slack/views/unlink_identity.py [hunk 1]
```diff
@@ -10,12 +11,18 @@
...
-from ..utils import logger, send_slack_response
+from ..utils import send_slack_response
 from . import build_linking_url as base_build_linking_url
 from . import never_cache
 
 SUCCESS_UNLINKED_MESSAGE = "Your Slack identity has been unlinked from your Sentry account."
 
+import logging
+
+from sentry.utils import metrics
```
Comment (cathteng) [inline src/sentry/integrations/slack/views/unlink_identity.py:22]: the importing looks kinda weird here, i think it should go above the const

File: src/sentry/integrations/slack/views/unlink_identity.py [hunk 2]
```diff
@@ -35,37 +42,91 @@ class SlackUnlinkIdentityView(BaseView):
...
+            logger.info("get_identity_success", extra={"integration_id": params["integration_id"]})
+            metrics.incr(
+                self._METRICS_SUCCESS_KEY + method, tags={"action": "get_identity"}, sample_rate=1.0
             )
+        except Http404:
+            logger.exception(
+                "get_identity_error", extra={"integration_id": params["integration_id"]}
+            )
+            metrics.incr(
+                self._METRICS_FAILURE_KEY + method, tags={"action": "get_identity"}, sample_rate=1.0
```
Comment (cathteng) [inline src/sentry/integrations/slack/views/unlink_identity.py:108]: should the metrics here be `post_identity`

**Review by cathteng — COMMENTED** [review]
i see there is some duplicate logic so i guess that's why it wasn't separated into `get` and `post` functions in the first place

File: src/sentry/integrations/slack/views/unlink_identity.py (hunk 2 above)
Comment (iamrajjoshi) [reply src/sentry/integrations/slack/views/unlink_identity.py]: i moved the function call to `dispatch` so we don't repeat it

File: src/sentry/integrations/slack/views/unlink_identity.py [hunk 3]
```diff
@@ -35,37 +41,69 @@ class SlackUnlinkIdentityView(BaseView):
...
-            integration_id=params["integration_id"],
+        try:
+            organization, integration, idp = get_identity_or_404(
+                ExternalProviders.SLACK,
+                request.user,
+                integration_id=params["integration_id"],
+            )
+            logger.info("get_identity_success", extra={"integration_id": params["integration_id"]})
+            metrics.incr(self._METRICS_SUCCESS_KEY + ".get_identity", sample_rate=1.0)
+            params.update({"organization": organization, "integration": integration, "idp": idp})
```
Comment (ykamo001) [inline src/sentry/integrations/slack/views/unlink_identity.py:67]: nit: these 3 lines should happen outside the try/catch

File: src/sentry/integrations/slack/views/unlink_identity.py [hunk 4]
```diff
@@ -35,37 +41,69 @@ class SlackUnlinkIdentityView(BaseView):
...
         )
 
-        if request.method != "POST":
-            return render_to_response(
-                "sentry/auth-unlink-identity.html",
-                request=request,
-                context={"organization": organization, "provider": integration.get_provider()},
-            )
+    def post(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
+        method = ".post."
```
Comment (ykamo001) [inline src/sentry/integrations/slack/views/unlink_identity.py:88]: nit: you probably don't need this as a variable

**Review by ykamo001 — APPROVED** [review]

File: src/sentry/integrations/slack/views/unlink_identity.py [hunk 5]
```diff
@@ -35,37 +41,69 @@ class SlackUnlinkIdentityView(BaseView):
...
+        params = kwargs["params"]
+        integration, idp = params["integration"], params["idp"]
 
         try:
             Identity.objects.filter(idp_id=idp.id, external_id=params["slack_id"]).delete()
         except IntegrityError:
             logger.exception("slack.unlink.integrity-error")
```
Comment (ykamo001) [inline src/sentry/integrations/slack/views/unlink_identity.py:95]: hopefully this captures/logs the ids we need/want during debugging

Comment (iamrajjoshi) [reply src/sentry/integrations/slack/views/unlink_identity.py:95]: lol it doesn't, ill add it

**Pull request merged**